src/mode.ts

```diff
@@ -53,6 +76,8 @@ function setupMode(defaults:LanguageServiceDefaultsImpl, modeId:string, language
 	disposables.push(new languageFeatures.DiagnostcsAdapter(defaults, modeId, worker));
 	disposables.push(monaco.languages.setLanguageConfiguration(modeId, richEditConfiguration));
 	disposables.push(monaco.languages.setTokensProvider(modeId, createTokenizationSupport(language)));
+
+	return client;

```

Comment:
Instead of the `client` you could export the `worker` function here. It already does the right thing. Also note that you should have a signature that takes `n` resources otherwise there framework might drop them again